### ISSUE TITLE
Use `actions/checkout@main` for all GitHub workflows

### ISSUE THREAD

**Issue by Panquesito7** [mt-mods/mt-mods#8]
### Description

Most of the GitHub workflows use `actions/checkout@v1` for LuaCheck mostly.
It is recommended to use the latest, stable version, which is `v4`.

This is not something critical, but it'd be great if we could update those.
What do you think about this? We can create PRs/commit for that. 🙂

**Comment by wsor4035** [comment]
this issue seems like a waste(unless your going to track specific repos statuses here), just make the prs to the relevant repos

edit: or tbh, trivial enough so just commit directly to them

**Comment by BuckarooBanzay** [comment]
> just commit directly to them

yeah, this is trivial enough :+1:

**Comment by OgelGames** [comment]
Would there be any reason not to use `actions/checkout@main`? Then we wouldn't need to update it again.

**Comment by S-S-X** [comment]
> Would there be any reason not to use `actions/checkout@main`? Then we wouldn't need to update it again.

Been broken before so there's that. Also as far as I know there's no really need to update from 2.x at all with current mt-mods use, new features are not used here at all.

To keep using somewhat tested release it is probably better to use v3 instead of following development.
If tested stability is concern then v2 would be best as it seems to be still getting bug fix releases, just no new features (or new bugs).

I've got nothing against update to v3 but I think it would still be good to use releases, we are not checkout action developers after all and therefore want somewhat stable platform instead of bleeding edge.

**Comment by Panquesito7** [comment]
If this sounds good, I'll commit directly to the repositories and update them.

**Comment by S-S-X** [comment]
Btw, actions/checkout@v4 been released some time ago.

**Comment by Panquesito7** [comment]
If this is still something that's wanted, I'd like to work on it. **EDIT:** Sorry for not doing this earlier by the way, got busy.

**Comment by Niklp09** [comment]
Please note: our official luacheck workflow uses `actions/checkout@master` not `v3/v4`.

**Comment by Panquesito7** [comment]
> Please note: our official luacheck workflow uses `actions/checkout@master` not `v3/v4`.

Should we use this in all repos, then? 🤔

**Comment by Niklp09** [comment]
> Should we use this in all repos, then? 🤔

yes, we should use **the same** workflow in all repos.

**Comment by S-S-X** [comment]
Probably better to add this here too:
actions/checkout@master is old abandoned branch (I think point after v2 and before v3).
Latest unstable / development branch is actions/checkout@main.

**Comment by Panquesito7** [comment]
> actions/checkout@master is old abandoned branch

Shouldn't `master` redirect to `main`?

**Comment by S-S-X** [comment]
> > actions/checkout@master is old abandoned branch
> 
> Shouldn't `master` redirect to `main`?

No, it is its own branch. I mean how'd you checkout or see history of it if it would redirect?
Not sure if GitHub redirects master to main if there's no master branch or if main was renamed from it instead of created from it but redirecting always would pretty much break git and redirecting based on certain rules would likely just be confusing.
For actions/checkout master is a real branch anyway, it is also fairly easy for you to check this yourself through https://github.com/actions/checkout

**Comment by Panquesito7** [comment]
Just noticed 👀 we should probably use `main` instead if you want to use the latest development version.

**Comment by OgelGames** [comment]
Just when you think you future-proofed something, you need to update it again...
